src/EiffelEvents.Net/Events/Edition-Lyon/Shared/EiffelEvent.cs

```diff
@@ -0,0 +1,127 @@
+﻿//    Copyright (c) 2021 ICT Cube, doWhile, and Eiffel Community collaborators.
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//        http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using EiffelEvents.Net.Common;
+using EiffelEvents.Net.Events.Core.Data;
+using EiffelEvents.Net.Events.Core.Links;
+using EiffelEvents.Net.Events.Edition_Lyon.Shared.Links;
+using EiffelEvents.Net.Exceptions;
+using Newtonsoft.Json;
+
+namespace EiffelEvents.Net.Events.Edition_Lyon.Shared
+{
+    public abstract record EiffelEvent<TData, TMeta, TLinks> :
+        Edition_Paris.Shared.EiffelEvent<TData, TMeta, TLinks>
+        where TData : EiffelData where TMeta : EiffelSharedMeta, new() where TLinks : EiffelLinks
+    {
+        #region Props
+
+        [JsonProperty("links")]
+        internal override IEiffelSerializedLinkCollection SerializedLinks { get; init; } =
+            new EiffelSerializedLinkCollection();
+
+        #endregion
+
+        #region Event Serialization / Deserialization
+
+        protected override void SerializeLinks()
```

Comment:
I think this method name could be changed for more descriptive name, as it not actuly serialize, rather than populate the `SerializedLinks `object with data.